wit/store.wit

```diff
@@ -0,0 +1,71 @@
+/// A keyvalue interface that provides eventually consistent key-value operations.
+/// 
+/// Each of these operations acts on a single key-value pair.
+/// 
+/// The value in the key-value pair is defined as a `u8` byte array and the intention is that it is
+/// the common denominator for all data types defined by different key-value stores to handle data,
+/// ensuring compatibility between different key-value stores. Note: the clients will be expecting
+/// serialization/deserialization overhead to be handled by the key-value store. The value could be
+/// a serialized object from JSON, HTML or vendor-specific data types like AWS S3 objects.
+/// 
+/// Data consistency in a key value store refers to the guarantee that once a write operation
+/// completes, all subsequent read operations will return the value that was written.
+/// 
+/// Any implementation of this interface must have enough consistency to guarantee "reading your
+/// writes." In particular, this means that you should never get a value that is older than the one
+/// you wrote, but you MAY get a newer value if one was written around the same time. For example,
+/// if a value at a given key is A, and you write B, then immediately read, you should get B. If
+/// something else writes C in quick succession, then you may get C.
```

Comment:
What does `you` refer to in the context of read-your-write consistency? 

I'd be more careful on defining the consistency guarantee here. 

In long-running process, I think `you` could refer to a client. That is the read-your-write consistency guarantees that once a write request is commited by a specific client, all subsequent read requests by the **same client** will reflect that write or any subsequent writes. 

In the context of serverless computing, I found it a bit more difficult to understand what read-your-write consistency really is. After reading the [session consistency document](https://learn.microsoft.com/en-us/azure/cosmos-db/consistency-levels#session-consistency) on Cosmos DB, I think the correct term for what `you` refers to is **logical session**. It refers to a context across multiple serverless function invocations, serving a single client. 

Also I would like to point out that after a write request is performed by a client, other clients may or may not immediately see the result due to the replication lag. This is the major difference between linearizable consistency and read-your-write consistency.

Reply:
Ok, I pushed all of that into the docs here. Let me know what you think

Reply:
Thanks, that indeed makes it much clearer! 

Would be nice if you could elaborate more on what client is in the context of serverless computing